lib/connect/tls.js

```diff
@@ -20,7 +20,8 @@ function buildBuilder (mqttClient, opts) {
 
   function handleTLSerrors (err) {
     // How can I get verify this error is a tls error?
-    if (opts.rejectUnauthorized) {
+    // Only throw error if it is not an connection refused error to reconnect
+    if (opts.rejectUnauthorized && err && err.code !== 'ECONNREFUSED') {
```

Comment:
@eiabea you can create a test-specific server with https://github.com/mqttjs/MQTT.js/blob/master/test/client.js#L242.